### DISCUSSION TITLE
GNSS/ IMU coupling Navego

### DISCUSSION THREAD

**Discussion by OmerZiaSiddiqui** [rodralez/NaveGo#154] [General]
Dr Rodrigo Gonzalez,

Sir, we are working on GNSS/ IMU coupling and had a chance to study your valuable paper (https://www.researchgate.net/publication/279239503) on the subject.

While understanding the Algorithm 2 at page 7, we feel that line 16 is contracting line 13 of the algorithm as stated below:

> 12: Guarantee that tG(1) < tS (1) =< tG(2)
> 13: Guarantee that tS (kS−1) < tG(kG) =< tS (kS )
> 14: Set i = 2
> 15: for k = 2 to kG do
> 16: while tS (i) =< tG(k) do

We will be thankful if you can share your expert opinion on the above, please.

Thanks
Omer Zia Siddiqui
PhD Scholar at NUST, Pakistan

**Comment by OmerZiaSiddiqui** [comment]
Further sir what is the purpose of line 12 and 13

**Comment by rodralez** [comment]
> Dr Rodrigo Gonzalez,
> 
> Sir, we are working on GNSS/ IMU coupling and had a chance to study your valuable paper (https://www.researchgate.net/publication/279239503) on the subject.
> 
> While understanding the Algorithm 2 at page 7, we feel that line 16 is contracting line 13 of the algorithm as stated below:
> 
> > 12: Guarantee that tG(1) < tS (1) =< tG(2)

Hello Omer,

This line of pseudo code awarded that the INS/GNSS system started when at least one GNSS measurement was available (tG(1)) 
 
> > 13: Guarantee that tS (kS−1) < tG(kG) =< tS (kS )

This line of pseudo code was to indicate that a new GNSS measurement  had to be available available ( tG(kG) ) before execution the Kalman filter.

The paper states that NaveGo master clock is GNSS clock but this is no longer true. Currently, the INS clock is NaveGo's master clock (for more details see https://github.com/rodralez/NaveGo/blob/d6e6b45279d78b7fef816660118143f6815b30a2/ins-gnss/ins_gnss.m#L237 ) 

In the main README.md file we inform that this and previous papers about NaveGo are outdated. We are looking for volunteers to update the documentation (see issue #108), but it seems nobody has time.

**Comment by OmerZiaSiddiqui** [comment]
Dr Rodralez

Sir thanks indeed for your valuable comments.

Let us know any assistance required for updating the documentation as you have mentioned. We can volunteer for that.

Thanks again

Omer

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by OmerZiaSiddiqui** [comment]
Dr Rodrigo Gonzalez

I noticed that I have mistakenly spelled your good name incorrectly in my last email. For sure it happened erroneously, please accept my pardon for that.

Thanks

Omer

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
